Makefile

```diff
@@ -30,15 +30,18 @@ build: ## Builds the all docker setup.
 	docker compose build
 
 run-devserver: ## Runs the dev-server
-	docker compose up angular-build -d
+# Add container label that it hasn't completed first run
+	COMPLETED_FIRST_RUN=0 docker compose up angular-build -d
 	$(MAKE) update.package
 	docker cp oppia-angular-build:/app/oppia/node_modules .
 	docker compose stop angular-build
-	docker compose up dev-server -d --no-deps
+	COMPLETED_FIRST_RUN=0 docker compose up dev-server -d --no-deps
 	$(MAKE) update.requirements
 	$(MAKE) run-offline
```

Comment:
change this to use start-devserver instead

Reply:
add the cmd that changes the `COMPLETED_FIRST_RUN` label after the start-devserver cmd

Reply:
@gp201 I think, we won't require to update `COMPLETED_FIRST_RUN` to update after start-devserver as it willbe automatically updated when health check completes successfully.

Reply:
@gp201 We can't update labels of docker container while it's running, thus I don't think we can implement updating docker label in when health checks are completed.

Some references:
https://stackoverflow.com/a/58063872

Reply:
What happens to `COMPLETED_FIRST_RUN, if a command in the dev-server container fails? Meaning, if the container successfully starts but stops because a command running inside the container fails

Reply:
Container is run twice in dev-server. When container is run successfully first time then it will set COMPLETED_FIRST_RUN to 1 second run, even if container fails to run successfully 2nd time.

Reply:
I'm sorry, but I didn't quite understand what you were trying to say. Could you please rephrase it for me?

Reply:
@gp201 It's quite complicated.
Let's say we have clean installation and we only executed `make build`.
- Now in first run line 38: `... docker compose up dev-server -d --no-deps` in `dev-server` will require internet and line 48: `docker compose up dev-server -d` in `start-devserver` will also require internet.
1. Now LABEL (COMPLETED_FIRST_RUN) stays false after line 38 is executed (regardless failed or successful run).
2. And LABEL is changed to `true`, as line 48 is executed (again even if it fails, label is set to true)
3. Problem: What is command 2 (ln 48) fails, then LABEL is still set to `true`, which shouldn't happen.
4. Solution: change cmd 1 (ln 38) to - `docker compose up dev-server -d`, so doing this internet isn't required for cmd 2 if cmd 1 is successfully run. And thus setting LABEL to true in cmd 2 will give required output. And if cmd 1 fails, line 39 (installing pip requirements) will fail and line 48 won't run, thus LABEL won't change to true.

Please, tell me if you still have any queries.

Reply:
After doing above, we won't require health check. And if health check fails, it most likely that problem isn't due to NO internet. 

Also, current flow:
![image](https://github.com/oppia/oppia/assets/93824505/9ff46479-9fa2-4306-9553-0f141e3a0422)